dev/NavigationView/NavigationView.h

```diff
@@ -272,6 +276,17 @@ class NavigationView :
     bool ShouldPreserveNavigationViewRS4Behavior();
     bool ShouldPreserveNavigationViewRS3Behavior();
 
+    bool NeedRearrangeOfTopElementsAfterOverflowSelectionChanged(int selectedOriginalIndex);
+    bool ShouldShowFocusVisual();
+    int LeftNavGetIndexFromItem(const winrt::IInspectable& data);
+    winrt::IInspectable LeftNavGetItemFromIndex(int index);
+    winrt::NavigationViewItemBase GetContainerForIndexPath(const winrt::IndexPath& indexPath);
+    bool IsContainerTheSelectedItemInTheSelectionModel(winrt::NavigationViewItemBase nvib);
+    bool IsContainerInOverflow(winrt::NavigationViewItemBase nvib);
+    void KeyboardFocusFirstItemFromItem(winrt::NavigationViewItemBase nvib);
+    void KeyboardFocusLastItemFromItem(winrt::NavigationViewItemBase nvib);
+    void ApplyCustomMenuItemContainerStyling(winrt::NavigationViewItemBase nvib, winrt::ItemsRepeater ir, int index);
+
     // Visual components
     tracker_ref<winrt::Button> m_paneToggleButton{ this };
```

Comment:
do we need to store and track all these visual components. can some of these be removed if we leverage the visual states instead ?

Reply:
I am pretty sure that some of this can be moved to visual states. However for this refactor I left the implementation as is in order to reduce the scope of required changes.